### ISSUE TITLE
What is the RAS alternative for firefox quantum?

### ISSUE THREAD

**Issue by ghost** [dillbyrne/random-agent-spoofer#584]
I love this addon but it seems because of some API issue @dillbyrne is stopping its development. Can some body point to a good alternative addons for RAS in firefox quantum.

@dillbyrne Please dont close this issue because its an important issue currently faced by RAS users. I for one searching for good alterantives for this add on and requires suggestion for this.

**Comment by gitbugged** [comment]
The only alternative I see at this point, is a massive uprising from Mozilla Firefox users, demanding that upstream implement RAS and privacy features upstream. Considering they just added Google search back into Firefox, I'm not sure we'll have much luck - but the world's greatest battles weren't won in a day. Keep the pressure on Mozilla and take the time to learn programming, all of us will need it in the near future. :+1:

**Comment by eemantsal** [comment]
@potholiday, there are several, from [this very simple one](https://addons.mozilla.org/es/firefox/addon/uaswitcher/) or [this other one](https://add0n.com/useragent-switcher.html#features), with more configuration options, to the rather complex [Umatrix](https://addons.mozilla.org/es/firefox/addon/umatrix/), that besides blocking scripts, CSS, cookies, media, etc, [can spoof your UA too](https://github.com/gorhill/uMatrix/wiki/Latest-user-agent-strings).

**Comment by Svetlana-T** [comment]
[agentx](https://addons.mozilla.org/en-US/firefox/addon/agentx/?src=search), too, has an option of random user agent it seems

**Comment by sheddup** [comment]
Whats an alternative with all the features of RAS, User Agent switching is just one of the features of RAS.

**Comment by neroux** [comment]
https://github.com/tarampampam/random-user-agent might be an option, it comes with a similar feature set.

**Comment by robinpaulson** [comment]
agentx is some shitty freeware licence though. no mention of reusing code, or any concrete statement of RMS' four freedoms. good suggestion, but i'll avoid it.

**Comment by mack318** [comment]
sad to see this go. its one of my favorites

**Comment by neroux** [comment]
@mack318 Have a look at the extension I mentioned earlier. The original developer hasnt been very active recently, but I managed to tweak a few things. It is not on Mozilla's extension page yet though.

**Comment by Laurentiu-Andronache** [comment]
neroux, that's a Chrome plugin right?

**Comment by neroux** [comment]
Yes and no. @tarampampam originally developed it for Chrome, but as Firefox now implemented Chrome's API (aka web extensions) most of them should work with no to little work. In this case there used to be an issue with a call to a deprecated method, but once that was fixed it seems to work fine now.

I have just published a first experimental version at https://addons.mozilla.org/firefox/addon/random_user_agent/

**Comment by Laurentiu-Andronache** [comment]
sweet!

your experimental version seems to work fine

if you could add to it all the user agents from here though:
http://techblog.willshouse.com/2012/01/03/most-common-user-agents/ 
from below “Get this list of the most common useragents as plain-text”, everything there!

please please? :D

**Comment by neroux** [comment]
User agents cant be "added" in that way, they are dynamically generated.

**Comment by Laurentiu-Andronache** [comment]
So when you select for example Internet Explorer 10, it won't always be the same? Then the extension is amazing. Any idea how many combinations it can generate?

**Comment by Laurentiu-Andronache** [comment]
Changing the user agent every minute also works!

**Comment by neroux** [comment]
> So when you select for example Internet Explorer 10, it won't always be the same? Then the extension is amazing. Any idea how many combinations it can generate?

The extension uses RandExp.js to generate strings based on a regex syntax. For IE10 it would be at https://github.com/tarampampam/random-user-agent/blob/master/extension/js/UAGenerator.js#L124

The combinations are only limited by how the regular expression is defined.

At the moment it still does require regular maintenance of the strings for Firefox and Chrome to add new versions and fade out older ones.

**Comment by ghost** [comment]
@neroux Do your addon have the curated list of useragents that we can select from like in RAS https://cloud.githubusercontent.com/assets/2903711/7324733/e0aa7140-eab0-11e4-889d-d1ceb7da9a0f.png . I like to use RAS to change the useragent to a specific one while searching a certain websites for example I manually select user agents  for `yahoo.com` as `FFOS - Flame - Firefox 47.0` . for `reddit.com` I use `WP8.1 - HTC 8X - IE 11` and use`whitelist` to search google.com and according to websites I select linux,windos,IOS,android etc useragents . Can we do that in your app?

**Comment by neroux** [comment]
@potholiday Please check out the earlier responses here. Technically it is no "my" extension though, I contributed some fixes and published it on addons.mozilla.org, thats it :). Dont want to take someone else's credit.

**Comment by sheddup** [comment]
I need something that replicates these settings, doesn't have to be a user agent switcher but this list of toggles was very convenient 
![untitled](https://user-images.githubusercontent.com/5480335/34170043-929bfa1a-e4d0-11e7-84d6-5014d45cfd08.png)

**Comment by neroux** [comment]
@sheddup I'd assume you could change most of these settings in about:config **manually**. At this moment in time (maybe this will change with future versions) extensions can, to my knowledge, not change these values anymore, as they do not have access to these user preference anymore as they did pre-57.

I could only imagine that an extension could currently semi-implement _some_ of them by injecting code into every website and (if possible) overriding that particular feature with dummy functions (blank iframes would be still an issue though).

**Comment by gotowf** [comment]
I've been using RAS for years to have this basic functionality that I was used to in Opera.

Now that Firefox dropped support for the extensions that made firefox useful to me, as there is basically zero chance Mozilla will offer this kind of features I decided to keep the extensions and replace Firefox with [Waterfox](https://www.waterfoxproject.org/). Waterfox is basically Firefox without the Mozilla nonsense: support for legacy extensions, disabled DRM (Encrypted Media Extensions), no pocket, no telemetry, no data collection, no startup profiling, allow unsigned extensions, no sponsored tiles and more.

Waterfox is currently building an alternative browser add-on store including legacy extensions with the possibility for author to make money with their addons. Hopefully RAS will be there.

**Comment by sheddup** [comment]
https://addons.mozilla.org/en-US/firefox/addon/webapi-manager/

**Comment by kekkc** [comment]
Why not push Mozilla to get our settings & functions back: https://github.com/dillbyrne/random-agent-spoofer/issues/544#issuecomment-347029473 ?

**Comment by classmate01** [comment]
@neroux 
Would love if you also add screen size spoofing in the future

**Comment by neroux** [comment]
@classmate01 Thanks for the suggestion, though may I suggest to continue issues of that sort at https://github.com/tarampampam/random-user-agent/issues? I wouldnt want to "spam" this one :)

As for that particular feature, I think it would be a bit outside of the extension's scope but @tarampampam would have the last say - I am merely a contributor :) Pull requests always welcome however

**Comment by SkyzohKey** [comment]
For people looking for screen spoofing, I've made a little user-script that you can install/download on Quantum from here: https://gist.github.com/SkyzohKey/1f8fe7c070cfe9e98f5dc9ec0c5caee2

**Comment by mdouchement** [comment]
A RAS for Firefox 57+ https://addons.mozilla.org/fr/firefox/addon/chameleon-ext/

https://www.ghacks.net/2018/06/18/chameleon-for-firefox-user-agent-and-data-spoofer/
